### PR TITLE
lang: semantic tests

### PR DESCRIPTION
**Pull request by shramee**
Semantic tests for set! and get! macros.
#924

### CODE REVIEW

File: crates/dojo-lang/src/semantics/test_data/get [hunk 1]
```diff
@@ -0,0 +1,408 @@
...
+            ),
+        ),
+        ty: <missing>,
+    },
+)
+
+//! > semantic_diagnostics
+error: Identifier not found.
+ --> get_inline_macro:2:96
+                let mut __get_macro_keys__ = array::ArrayTrait::new();serde::Serde::serialize(@key, ref __get_macro_keys__);
```
Comment (tarrencev) [inline crates/dojo-lang/src/semantics/test_data/get:382]: Can we improve formatting here? i.e. new line. I guess it's not possible to define key / suppress this? It adds noise to the test

Comment (shramee) [reply crates/dojo-lang/src/semantics/test_data/get]: It's just showing the whole line where stuff happens. I think most straightforward is to add some line breaks in macro expansion. Which is what I've done in the latest commit 😉

File: crates/dojo-lang/src/semantics/test_data/get [hunk 2]
```diff
@@ -0,0 +1,408 @@
...
+            ),
+        ),
+        ty: <missing>,
+    },
+)
+
+//! > semantic_diagnostics
+error: Identifier not found.
+ --> get_inline_macro:3:26
+serde::Serde::serialize(@key, ref __get_macro_keys__);
```
Comment (tarrencev) [inline crates/dojo-lang/src/semantics/test_data/get:382]: If we add key and world to setup and intern dojo-core in the semantic db i think these diagnostics will go away

Comment (shramee) [reply crates/dojo-lang/src/semantics/test_data/get:382]: Aah, I misunderstood...
Yeah, I can look into it.

Comment (shramee) [reply crates/dojo-lang/src/semantics/test_data/get:382]: To get rid of all diagnostic, we actually need the whole dojo crate to have it work.
Or mock stuff we need.

Comment (shramee) [reply crates/dojo-lang/src/semantics/test_data/get]: Done
# 🫡

**Review by tarrencev — APPROVED** [review]

**Pull request merged**